Sources/Workspace/Workspace.swift

```diff
@@ -980,25 +964,30 @@ public class Workspace {
     private func validateEditedPackages() throws {
         let managedDependencies = try self.managedDependencies.load()
         for dependency in managedDependencies.values {
-
-            let dependencyPath: AbsolutePath
-
             switch dependency.state {
             case .checkout: continue
             case .edited:
-                dependencyPath = editablesPath.appending(dependency.subpath)
-            case .unmanaged(let path):
-                dependencyPath = path
-            }
-
-            // If some edited dependency has been removed, mark it as unedited.
-            if !fileSystem.exists(dependencyPath) {
-                try unedit(dependency: dependency, forceRemove: true)
-                // FIXME: Use diagnosics engine when we have that.
-                delegate.warning(message: "\(dependencyPath.asString) was being edited but has been removed, falling back to original checkout.")
+                let dependencyPath = path(for: dependency)
+                
+                // If some edited dependency has been removed, mark it as unedited.
+                if !fileSystem.exists(dependencyPath) {
+                    try unedit(dependency: dependency, forceRemove: true)
+                    // FIXME: Use diagnosics engine when we have that.
+                    delegate.warning(message: "\(dependency.subpath.asString) was being edited but has been removed, falling back to original checkout.")
+                }
             }
         }
     }
+    
+    /// Returns the location of the dependency
+    private func path(for dependency: ManagedDependency) -> AbsolutePath {
```

Comment:
I was following the *Omit needless words.* rule of the API Design Guidelines. Let me know if you still want me to rename it. Here's the rule btw:

## Omit needless words.

Every word in a name should convey salient information at the use site.

More words may be needed to clarify intent or disambiguate meaning, but those that are redundant with information the reader already possesses should be omitted. In particular, omit words that merely repeat type information.

```
public mutating func removeElement(_ member: Element) -> Element?
allViews.removeElement(cancelButton)
```

In this case, the word Element adds nothing salient at the call site. This API would be better:

```
public mutating func remove(_ member: Element) -> Element?
allViews.remove(cancelButton) // clearer
```

Occasionally, repeating type information is necessary to avoid ambiguity, but in general it is better to use a word that describes a parameter’s role rather than its type. See the next item for details.

My reasoning was that the type information was enough information to suggest we are looking for the path of a dependency. Adding dependency in the name could be interpreted as a "needless word" following this rule.